commons-vfs2/src/main/java/org/apache/commons/vfs2/impl/DefaultFileSystemManager.java

```diff
@@ -712,13 +713,52 @@ public FileObject resolveFile(final FileObject baseFile, final String uri, FileS
             // An absolute URI - locate the provider
             final FileProvider provider = providers.get(scheme);
             //In the case of SFTP set the path from root if the param is presented in URL
-            if (provider instanceof SftpFileProvider && "true".equals(queryParam.get(SftpConstants
-                    .SFTP_PATH_FROM_ROOT))) {
+            if (provider instanceof SftpFileProvider) {
                 if (fileSystemOptions == null) {
                     fileSystemOptions = new FileSystemOptions();
                 }
-                ((SftpFileSystemConfigBuilder) ((provider).getConfigBuilder())).setUserDirIsRoot
-                        (fileSystemOptions, false);
+                if ("true".equals(queryParam.get(SftpConstants.SFTP_PATH_FROM_ROOT))) {
+                    ((SftpFileSystemConfigBuilder) (((SftpFileProvider) provider).getConfigBuilder()))
+                            .setUserDirIsRoot(fileSystemOptions, false);
+                }
+
+                String proxyHost = queryParam.get(SftpConstants.PROXY_SERVER);
+
+                if (proxyHost != null && !proxyHost.isEmpty()) {
+                    String proxyPortStr = queryParam.get(SftpConstants.PROXY_PORT);
+                    int proxyPort = 8080;
+                    if (proxyPortStr != null) {
+                        try {
+                            proxyPort = Integer.parseInt(proxyPortStr);
+                        } catch (NumberFormatException e) {
+                            proxyPort = 8080;
```

Comment:
Shall we introduce a log here too, indicating the specification of an invalid value?